electron/src/main/main.ts

```diff
@@ -112,9 +134,33 @@ const createWindow = async () => {
   new AppUpdater();
 };
 
-/**
- * Add event listeners...
- */
+ipcMain.handle('get-user-data-path', () => electronApp.getPath('userData'));
+
+ipcMain.handle(
+  'open-realm',
+  async (
+    _: Electron.IpcMainInvokeEvent,
+    config: Realm.Configuration
+  ): Promise<boolean | null> => {
+    console.log('data on main', config);
+    if (!config?.sync?.user?.id) {
+      // @ts-ignore
+      config.sync.user = await logInRealm();
+    }
+    console.log('user', config?.sync?.user?.id);
```

Comment:
this weird code was added b.c it seems that electron isn't able to pass the logged in use from the renderer process to here on main. spent a bunch of time debugging and trying to get way to pass it, but was not successful. 

here should be replaced w logging the user in over IPC, and then passing those credentials to this method.